src/main/java/org/killbill/billing/plugin/stripe/StripePaymentPluginApi.java

```diff
@@ -375,8 +380,14 @@ public void deletePaymentMethod(final UUID kbAccountId, final UUID kbPaymentMeth
         return super.getPaymentMethods(kbAccountId, false, properties, context);
     }
 
-    private void syncPaymentMethods(final UUID kbAccountId, final Iterable<? extends HasId> stripeObjects, final Map<String, StripePaymentMethodsRecord> existingPaymentMethodByStripeId, final CallContext context) throws PaymentApiException, SQLException {
+    private void syncPaymentMethods(final UUID kbAccountId, final Iterable<? extends HasId> stripeObjects, final Map<String, StripePaymentMethodsRecord> existingPaymentMethodByStripeId, final Set<String> stripeObjectsTreated, final CallContext context) throws PaymentApiException, SQLException {
         for (final HasId stripeObject : stripeObjects) {
+            if (stripeObjectsTreated.contains(stripeObject.getId())) {
+                continue;
+            } else {
```

Comment:
Do we need the `else` -- seems like if `! stripeObjectsTreated.contains(stripeObject.getId())` we want to execute the `else` and whatever comes after?

Reply:
Fixed in f879509f967473b1a3eca9941dd3a523ed8c002e.